action/governance/withdrawFunds.go

```diff
@@ -99,37 +99,44 @@ func runWithdraw(ctx *action.Context, signedTx action.RawTx) (bool, action.Respo
 		}
 		return false, result
 	}
+
 	fundStore := ctx.ProposalMasterStore.ProposalFund
 	currentFundsForProposal := fundStore.GetCurrentFundsForProposal(proposal.ProposalID)
-	// if funding goal is reached or there is still time for funding
-	if currentFundsForProposal.BigInt().Cmp(proposal.FundingGoal.BigInt()) >= 0 || ctx.Header.Height <= proposal.FundingDeadline {
-		ctx.Logger.Error("Proposal does not meet withdraw requirement", withdrawProposal.ProposalID)
-		result := action.Response{
-			Events: action.GetEvent(withdrawProposal.Tags(), "withdraw_proposal_does_not_meet_withdraw_requirement"),
-			Log:    governance.ErrProposalWithdrawNotEligible.Marshal(),
+	// if outcome is not cancelled or insufficient funds
+	if proposal.Outcome != governance.ProposalOutcomeCancelled && proposal.Outcome != governance.ProposalOutcomeInsufficientFunds {
```

Comment:
Did you miss the 2nd scenario?
The 1st scenario: (proposal.Outcome != governance.ProposalOutcomeCancelled && currentFunds < fundingGoal && currentHeight > fundingDeadline)
The 2nd scenario: (proposal.Outcome == governance.ProposalOutcomeCancelled)

Reply:
if cancelled then there should be no error and we don't need to change status, outcome, state